modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1462,6 +1470,63 @@ else if (Objects.equals(
 		}
 	}
 
+	private void _addLocalizedObjectFieldValues(
+		DynamicObjectDefinitionLocalizationTable
+			dynamicObjectDefinitionLocalizationTable,
+		long objectEntryId, Map<String, Serializable> values) {
+
+		Column<DynamicObjectDefinitionLocalizationTable, Long>
+			foreignKeyColumn =
+				dynamicObjectDefinitionLocalizationTable.getForeignKeyColumn();
+
+		List<Object[]> localizedValues = ObjectMapperUtil.readValue(
+			List.class,
+			(Serializable)objectEntryPersistence.dslQuery(
+				DSLQueryFactoryUtil.select(
+					ArrayUtil.append(
+						_getSelectExpressions(
+							dynamicObjectDefinitionLocalizationTable),
+						dynamicObjectDefinitionLocalizationTable.
+							getLanguageIdColumn())
+				).from(
+					dynamicObjectDefinitionLocalizationTable
+				).where(
+					foreignKeyColumn.eq(objectEntryId)
+				)));
+
+		if (ListUtil.isEmpty(localizedValues)) {
+			return;
+		}
+
+		List<Column<DynamicObjectDefinitionLocalizationTable, ?>>
+			objectFieldColumns =
+				dynamicObjectDefinitionLocalizationTable.
+					getObjectFieldColumns();
+
+		for (int i = 0; i < objectFieldColumns.size(); i++) {
+			Map<String, String> localizedObjectFieldValue = new HashMap<>();
+
+			for (Object[] localizedValue : localizedValues) {
+				Object value = localizedValue[i];
+
+				if (Validator.isNull(value)) {
+					continue;
+				}
+
+				localizedObjectFieldValue.put(
+					String.valueOf(localizedValue[objectFieldColumns.size()]),
```

Comment:
@seltong I believe we should create a variable to avoid confusions, something like:

```java 
int laguageIdColumnPosition = objectFieldColumns.size();
```